### PR TITLE
Ch 1093 add task version transformation

### PR DESCRIPTION
**Pull request by kkoehn**
Add the ability to transform tasks between different proforma versions

requirement to fix [codeharbor/1093](https://github.com/openHPI/codeharbor/issues/1093) and [codeharbor/1094](https://github.com/openHPI/codeharbor/issues/1094)

### CODE REVIEW

File: lib/proformaxml/services/exporter.rb [hunk 1]
```diff
@@ -99,13 +102,6 @@ def tests(xml)
     end
 
     # ms-placeholder only necessary for version 2.0 where model-solutions were mandatory
```
Comment (Dome-GER) [inline lib/proformaxml/services/exporter.rb:104]: This comment can be removed as well, I'd say.

File: lib/proformaxml/services/importer.rb [hunk 2]
```diff
@@ -18,23 +19,20 @@ def initialize(zip:, expected_version: nil)
...
+
       errors = validate
 
       raise PreImportValidationError.new(errors) if errors.any?
 
-      @pro_ns = proforma_namespace
       @task_node = @doc.xpath("/#{@pro_ns}:task")
 
       set_data
+      # if version != latest_version
```
Comment (Dome-GER) [inline lib/proformaxml/services/importer.rb:33]: Huh?

**Review by Dome-GER — CHANGES_REQUESTED** [review]

File: lib/proformaxml/services/importer.rb (hunk 2 above)
Comment (kkoehn) [reply lib/proformaxml/services/importer.rb]: added the planned functionality

File: lib/proformaxml/services/version_and_namespace_extractor.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    private
+
+    def extract_schema_and_version
+      namespace_regex = /^urn:proforma:v(\d.*)$/
+      potential_namespaces = @doc.namespaces.filter do |_, href|
+        href.match? namespace_regex
+      end
+      return nil unless potential_namespaces.length == 1
```
Comment (Dome-GER) [inline lib/proformaxml/services/version_and_namespace_extractor.rb:23]:
```suggestion
      return unless potential_namespaces.length == 1
```
🤓

**Review by Dome-GER — APPROVED** [review]
I very much like the refactoring into services. It also seems to be tested extensively, I did not dive into all the nitty-gritty details, since I'm lacking some domain knowledge. The overall design decisions look very good,

**Pull request merged**